doc: Improve scripted-diff developer docs (#14731)
bitcoin/bitcoin: doc/developer-notes.md

    - `-BEGIN VERIFY SCRIPT-`
    - `-END VERIFY SCRIPT-`

The scripted-diff is verified by the tool `test/lint/commit-script-check.sh`. The tool's default behavior when supplied
with a commit is to verify all scripted-diffs since the beginning of time. For development, it might be more convienient

> to verify all scripted-diffs in commits since `origin/master`:

Reviewer: ```suggestion
to verify all scripted-diffs in a range `A..B`, for example:
```

nit: Not sure if worth changing, but it should accept any range.

    - `-BEGIN VERIFY SCRIPT-`
    - `-END VERIFY SCRIPT-`

The scripted-diff is verified by the tool `test/lint/commit-script-check.sh`. The tool's default behavior when supplied
with a commit is to verify all scripted-diffs from the beginning of time up to said commit. Internally, the tool passes
the first supplied argument to `git rev-list --reverse` to determine which commits to verify script-diffs for, ignoring
commits that don't conform to the commit message format described above.

> For development, it might be more convienient to verify all scripted-diffs in a range `A..B`, for example:

Reviewer 2: Convenient :-)
